src/qmapshack/gis/prj/IGisProject.cpp

```diff
@@ -1141,6 +1145,11 @@ void IGisProject::sortItems()
     }
 
     addChildren(items);
+    if(projectFilter != nullptr)
+    {
+        projectFilter->showLineEdit(&projectSearch);
+    }
```

Comment:
Why do we have to destroy and instantiate the `CSearchLineEdit` widget on each sorting action?

Reply:
`CSearchLineEdit` is placed onto the `CProjectFilterItem`. However, when sorting the project, every child is removed and qt automatically destroys the widget, since the `QTreeWidgetItem` that it is placed on isn't part of the `QTreeWidget` anymore. Thus we have to recreate it after every sorting.